tests/test_spidermiddleware_urllength.py

```diff
@@ -20,22 +20,36 @@ def setUp(self):
         self.stats = crawler.stats
         self.mw = UrlLengthMiddleware.from_settings(settings)
 
+        self.maxlength_infinite = 0
+        settings_infinite = Settings({'URLLENGTH_LIMIT': self.maxlength_infinite})
+
+        crawler_infinite = get_crawler(Spider)
+        self.spider_infinite = crawler_infinite._create_spider('foo_infinite')
+        self.stats_infinite = crawler_infinite.stats
+        self.mw_infinite = UrlLengthMiddleware.from_settings(settings_infinite)
         self.response = Response('http://scrapytest.org')
+
         self.short_url_req = Request('http://scrapytest.org/')
         self.long_url_req = Request('http://scrapytest.org/this_is_a_long_url')
         self.reqs = [self.short_url_req, self.long_url_req]
```

Comment:
This `setUp` method was written for a test case that was testing a single scenario. Now that the test is going to cover more than that, I think it makes sense to, rather than extending `setUp`, to move stuff out of `setUp`, and leave in `setUp` only whatever is common to all scenarios (if anything).